src/top-bar/selection/selection-provider.js

```diff
@@ -0,0 +1,135 @@
+    code: code,
+    displayName: code,
+})
+
+const initialWorkflowValue = (workflows, id) => {
+    if (id) {
+        /*
+         * Auto select workflow with query param id
+         * default to empty object if `find` returns undefined in case the
+         * workflow with the id from the url is not available to the user
+         */
+        return workflows.find(workflow => workflow.id === id) || {}
+    }
+    if (workflows.length === 1) {
+        // auto-select if user only has one workflow
+        return workflows[0]
+    }
+    return {}
+}
+
+const ACTIONS = {
+    SET_OPENED_SELECT: 'SET_OPENED_SELECT',
+    CLEAR_ALL: 'CLEAR_ALL',
+    SELECT_WORKFLOW: 'SELECT_WORKFLOW',
+    SELECT_PERIOD: 'SELECT_PERIOD',
+    SELECT_ORG_UNIT: 'SELECT_ORG_UNIT',
+}
+
+const reducer = (state, { type, payload }) => {
+    switch (type) {
+        case ACTIONS.SET_OPENED_SELECT:
+            return {
+                ...state,
+                openedSelect: payload,
+            }
+        case ACTIONS.CLEAR_ALL:
+            return {
+                openedSelect: '',
+                workflow: {},
+                period: {},
+                orgUnit: {},
+            }
+        case ACTIONS.SELECT_WORKFLOW:
+            return {
+                openedSelect: '',
+                workflow: payload,
+                period: {},
+                orgUnit: {},
+            }
+        case ACTIONS.SELECT_PERIOD:
+            return {
+                ...state,
+                openedSelect: '',
+                period: payload,
+                orgUnit: {},
+            }
+        case ACTIONS.SELECT_ORG_UNIT:
+            return {
+                ...state,
+                openedSelect: '',
+                orgUnit: payload,
+            }
+        default:
+            return state
+    }
+}
+
+const SelectionProvider = ({ children }) => {
+    const [query, setQuery] = useQueryParams({
+        wf: StringParam,
+        pe: StringParam,
+        ou: StringParam,
+    })
+    const { dataApprovalWorkflows } = useCurrentUser()
+    const [{ openedSelect, workflow, period, orgUnit }, dispatch] = useReducer(
+        reducer,
+        {
+            openedSelect: '',
+            workflow: initialWorkflowValue(dataApprovalWorkflows, query.wf),
+            // TODO: the initial value for period should take into account the inital
+            // workflow value, it should be cleared if no valid workflow is found
+            period: query.pe ? parsePeriodCode(query.pe) : {},
+            // TODO: same as period, but orgUnit should also be cleared if period is
+            // unset/invalid
+            orgUnit: query.ou ? { id: query.ou } : {},
+        }
+    )
+    const providerValue = {
+        workflow,
+        period,
+        orgUnit,
+        openedSelect,
+        clearAll: () =>
+            dispatch({
+                type: ACTIONS.CLEAR_ALL,
+                payload: initialWorkflowValue(dataApprovalWorkflows),
```

Comment:
The `CLEAR_ALL` handling section in the reducer does not use the payload.
Either the reducer is wrong, or the action creator is wrong

Reply:
After reading this a bit more carefully, I think it does make sense to use the payload in case there's only one workflow, but then the reducer needs to be fixed

Reply:
Good find, this was a mistake.